frontend/javascripts/oxalis/view/action-bar/share_modal_view.js

```diff
@@ -126,14 +126,19 @@ class ShareModalView extends PureComponent<Props, State> {
   };
 
   handleOk = async () => {
-    await updateTeamsForSharedAnnotation(
-      this.props.annotationType,
-      this.props.annotationId,
-      this.state.sharedTeams.map(team => team.id),
-    );
-    Toast.success(messages["annotation.shared_teams_edited"]);
+    await editAnnotation(this.props.annotationId, this.props.annotationType, {
+      visibility: this.state.visibility,
+    });
+
+    if (this.state.visibility !== "Private") {
```

Comment:
Does the backend set the shared teams to `[]` when the visibility is set to private?